frontend/streamlit_app.py

```diff
@@ -0,0 +1,159 @@
+        )
+
+    if "chat_history" not in st.session_state:
+        st.session_state.chat_history = []
+    if "metadata" not in st.session_state:
+        st.session_state.metadata = {}
+
+    if not st.session_state.chat_history:
+        st.session_state.chat_history.append({
+            "content": "Hi, I am the OpenROAD assistant. Type your query about OpenROAD",
+            "role": "ai",
+        })
+
+    for message in st.session_state.chat_history:
+        with st.chat_message(message["role"]):
+            st.markdown(message["content"])
+
+    user_input = st.chat_input("Enter your queries ...")
+
+    if user_input:
+        st.session_state.chat_history.append({"content": user_input, "role": "user"})
+
+        with st.chat_message("user"):
+            st.markdown(user_input)
+
+        response, sources = response_generator(user_input)
+        if response is not None:
+            response_buffer = ""
+
+            with st.chat_message("ai"):
+                message_placeholder = st.empty()
+
+                response_buffer = ""
+                for chunk in response.split(' '):
+                    response_buffer += chunk + ' '
+                    if chunk.endswith('\n'):
+                        response_buffer += ' '
+                    message_placeholder.markdown(response_buffer)
+                    time.sleep(0.05)
+
+                message_placeholder.markdown(response_buffer)
+            st.session_state.chat_history.append(
+                {"content": response_buffer, "role": "ai"})
+
+            if sources:
+                with st.expander("Sources:"):
+                    try:
+                        print(sources)
+                        if isinstance(sources, str):
+                            cleaned_sources = sources.replace("{", "[").replace("}", "]")
+                            parsed_sources = ast.literal_eval(cleaned_sources)
+                            print(parsed_sources)
+                        else:
+                            parsed_sources = sources
+                        if isinstance(parsed_sources, (list, set)):
+                            sources_list = "\n".join(
+                                f"- [{link}]({link})"
+                                for link in parsed_sources
+                                if link.strip()
+                            )
+                            st.markdown(sources_list)
+                        else:
+                            st.markdown("No valid sources found.")
+                    except (ValueError, SyntaxError) as e:
+                        st.markdown(f"Failed to parse sources: {e}")
+
+    question_dict = {
+        interaction["content"]: i
+        for i, interaction in enumerate(st.session_state.chat_history)
+        if interaction["role"] == "user"
+    }
+    if question_dict and os.getenv("FEEDBACK_SHEET_ID"):
```

Comment:
if `FEEDBACK_SHEET_ID` is necessary, make it a if condition within the `if question_dict` logic. Otherwise you will just not display anything if its not provided

Reply:
That's what I intended. If the user hasn't set up the feedback sheet ID, there's no need to display a feedback button and the feedback side page.